dashboard/src/main/resources/templates/component.ftl

```diff
@@ -110,8 +110,28 @@
     </#if>
 
     <h2>Dependencies</h2>
-    <pre class="dependency-tree">${dependencyTree}</pre>
-    
+
+    <#macro formatDependencyNode tree currentNode parent>
+      <#-- tree: ListMultimap<DependencyPath, DependencyPath>, currentNode: DependencyPath -->
+      <#if currentNode.size() gt 0>
+        <#-- root node does not have any leaf -->
+        <#if parent.size() gt 0>
+          <#assign label = 'parent: ' + parent.getLeaf() />
+        <#else>
+          <#assign label = 'root' />
+        </#if>
+        <p class="DEPENDENCY_TREE_NODE" title="${label}">${currentNode.getLeaf()}</p>
```

Comment:
I don't think you need the p here. You can just use the li element.

Reply:
We need an HTML tag to hold the label (title) to show the mouse-hover. Adding title to li would make the scope of the li too big.